osu.Game/Rulesets/Mods/ModAutoplay.cs

```diff
@@ -5,30 +5,31 @@
 using osu.Game.Beatmaps;
 using osu.Game.Graphics;
 using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Replays;
 using osu.Game.Rulesets.Scoring;
 using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Rulesets.Mods
 {
-    public abstract class ModAutoplay<T> : ModAutoplay, IApplicableToRulesetContainer<T>
+    public class ModAutoplay<T> : ModAutoplay, IApplicableToRulesetContainer<T>, IApplicableFailOverride
         where T : HitObject
     {
-        protected abstract Score CreateReplayScore(Beatmap<T> beatmap);
+        protected virtual Score CreateReplayScore(Beatmap<T> beatmap) => new Score { Replay = new Replay() };
 
-        public virtual void ApplyToRulesetContainer(RulesetContainer<T> rulesetContainer)
-        {
-            rulesetContainer.SetReplay(CreateReplayScore(rulesetContainer.Beatmap)?.Replay);
-        }
+        public override bool HasImplementation => GetType().GenericTypeArguments.Length == 0;

```

Comment:
I think this should be done at a base level, no? I foresee us having more generic mod implementations like this.

Reply:
let's revisit this when another generic mod comes up